components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/processor/SAMLSubjectQueryProcessor.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.identity.query.saml.processor;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.opensaml.saml.saml2.core.Issuer;
+import org.opensaml.saml.saml2.core.RequestAbstractType;
+import org.opensaml.saml.saml2.core.Response;
+import org.opensaml.saml.saml2.core.Subject;
+import org.opensaml.saml.saml2.core.SubjectQuery;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.core.model.SAMLSSOServiceProviderDO;
+import org.wso2.carbon.identity.query.saml.QueryResponseBuilder;
+import org.wso2.carbon.identity.query.saml.handler.SAMLAttributeFinder;
+import org.wso2.carbon.identity.query.saml.handler.UserStoreAttributeFinder;
+import org.wso2.carbon.identity.query.saml.util.SAMLQueryRequestUtil;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This calss is used to process common elements of
+ * AttributeQuery,AuthnQuery,
+ * AuthorizationDecisionQuery,SubjectQuery messages.This class is the parent of
+ * given class list.
+ */
+public class SAMLSubjectQueryProcessor implements SAMLQueryProcessor {
+
+    final static Log log = LogFactory.getLog(SAMLSubjectQueryProcessor.class);
+
+    /**
+     * This method used to generate response object according to subject
+     *
+     * @param request assertion request message
+     * @return Response container of one or more assertions
+     */
+    public Response process(RequestAbstractType request) {
+
+        SubjectQuery query = (SubjectQuery) request;
+        String user = getUserName(query.getSubject());
```

Comment:
extracting issuer info from the request is redundant in all classes extending SAMLQueryProcessor. Better to use bean object + implement a protected method at the super class level to remove this redundancy.